packages/vscode-extension/src/extension.ts

```diff
@@ -824,6 +829,27 @@ async function runBackgroundAsyncTasks(
   context: vscode.ExtensionContext,
   isTeamsFxProject: boolean
 ) {
+  await exp.initialize(context);
+  TreatmentVariableValue.welcomeViewStyle = (await exp
+    .getExpService()
+    .getTreatmentVariableAsync(
+      TreatmentVariables.VSCodeConfig,
+      TreatmentVariables.WelcomeView,
+      true
+    )) as string | undefined;
+  await vscode.commands.executeCommand(
+    "setContext",
+    "fx-extension.isExistingUser",
+    isExistingUser !== "no"
+  );
+  if (TreatmentVariableValue.welcomeViewStyle === "A") {
+    await vscode.commands.executeCommand("setContext", "fx-extension.welcomeViewTreatment", true);
+    await vscode.commands.executeCommand("setContext", "fx-extension.welcomeViewA", true);
```

Comment:
If the group is re-randomized, it may happen that both "fx-extension.welcomeViewA" and "fx-extension.welcomeViewB" contexts are true. Can we also set another context to avoid this situation?

Reply:
The context will be cleared when VS Code starts.